mkosi/__init__.py

```diff
@@ -5947,11 +5956,19 @@ def run_qemu(args: CommandLineArguments) -> None:
         # after it is created.
         cmdline += ["-nic", f"tap,script=no,downscript=no,ifname={ifname}"]
 
-    cmdline += args.cmdline
+    with contextlib.ExitStack() as stack:
+        if args.ephemeral:
+            f = stack.enter_context(copy_image_temporary(src=args.output, dir=os.path.dirname(args.output)))
+            fname = f.name
+        else:
+            fname = args.output
+
+        cmdline += ["-drive", f"format={'qcow2' if args.qcow2 else 'raw'},file={fname},if=virtio"]
+        cmdline += args.cmdline
 
-    print_running_cmd(cmdline)
+        print_running_cmd(cmdline)
 
-    run(cmdline, execvp=True)
+        run(cmdline, execvp=True)
```

Comment:
I haven't tested this, but does this clean up properly? From the top of my head I don't know whether the ExitStack can be / will be worked through if we exec here.

Reply:
Shit, I didn't consider that. Maybe we should get rid of the execvp?

Reply:
With a very quick test, this doesn't seem to cause any major problems.

Reply:
That could be done, I guess. Although I don't know whether there was any other intent behind execing here, besides "mkosi is done and doesn't need to stay around".

Reply:
It makes signal handling somewhat more awkward as there's two processes running now. I'll need to verify this doesn't break CTRL+C when using the qemu gui.

Reply:
This works fine, we already handle the two process scenario in run() correctly. I think we can go ahead and change this. I'll do the same change for mkosi boot